core/tee/tee_enc_fs_key_manager.c

```diff
@@ -0,0 +1,362 @@
+static uint8_t string_for_ssk_gen[] = "ONLY_FOR_ENC_FS_SSK";
+
+
+static TEE_Result fek_crypt(TEE_OperationMode mode,
+			uint8_t *key, int size)
+{
+	TEE_Result res;
+	uint8_t *ctx = NULL;
+	size_t ctx_size;
+	uint8_t dst_key[ENC_FS_FEK_SIZE];
+
+	if (key == NULL)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (size != ENC_FS_FEK_SIZE)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (enc_fs_ssk.is_init == 0)
+		return TEE_ERROR_CANCEL;
+
+	res = crypto_ops.cipher.get_ctx_size(TEE_ALG_AES_ECB_NOPAD, &ctx_size);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	ctx = malloc(ctx_size);
+	if (ctx == NULL)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	res = crypto_ops.cipher.init(ctx, TEE_ALG_AES_ECB_NOPAD,
+			mode, enc_fs_ssk.key, ENC_FS_SSK_SIZE,
+			NULL, 0, NULL, 0);
+	if (res != TEE_SUCCESS)
+		goto exit;
+
+	res = crypto_ops.cipher.update(ctx, TEE_ALG_AES_ECB_NOPAD,
+			mode, true, key, size, dst_key);
+	if (res != TEE_SUCCESS)
+		goto exit;
+
+	crypto_ops.cipher.final(ctx, TEE_ALG_AES_ECB_NOPAD);
+
+	memcpy(key, dst_key, sizeof(dst_key));
+
+exit:
+	free(ctx);
+
+	return res;
+}
+
+static TEE_Result generate_fek(uint8_t *key, uint8_t len)
+{
+	return crypto_ops.prng.read(key, len);
+}
+
+static TEE_Result generate_iv(uint8_t *iv, uint8_t len)
+{
+	return crypto_ops.prng.read(iv, len);
+}
+
+static TEE_Result generate_ssk(uint8_t *ssk, uint32_t ssk_size,
+			uint8_t *huk, uint32_t huk_size,
+			uint8_t *message, uint32_t message_size)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	uint8_t *ctx = NULL;
+	size_t hash_ctx_size = 0;
+
+	if (ssk == NULL || huk == NULL || message == NULL)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	res = crypto_ops.mac.get_ctx_size(ENC_FS_HMAC_ALG, &hash_ctx_size);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	ctx = malloc(hash_ctx_size);
+	if (ctx == NULL)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	res = crypto_ops.mac.init(ctx, ENC_FS_HMAC_ALG, huk, huk_size);
+	if (res != TEE_SUCCESS)
+		goto exit;
+
+	res = crypto_ops.mac.update(ctx, ENC_FS_HMAC_ALG,
+			message, message_size);
+	if (res != TEE_SUCCESS)
+		goto exit;
+
+	res = crypto_ops.hash.final(ctx, ENC_FS_HMAC_ALG, ssk, ssk_size);
+	if (res != TEE_SUCCESS)
+		goto exit;
+
+	res = TEE_SUCCESS;
+
+exit:
+	free(ctx);
+	return res;
+}
+
+static TEE_Result enc_fs_init_key_manager(void)
+{
+	int res = TEE_SUCCESS;
+
+	if (enc_fs_ssk.is_init == 0) {
+		struct tee_hw_unique_key huk;
+		uint8_t chip_id[CHIP_ID_LENGTH];
+		uint8_t *message;
+
+		/* Secure Storage Key Generation:
+		 *
+		 *     SSK = HMAC(HUK, message)
+		 *     message := concatenate(chip_id, static string)
+		 * */
+
+		message = malloc(sizeof(chip_id) + sizeof(string_for_ssk_gen));
+		if (message == NULL) {
+			res = TEE_ERROR_OUT_OF_MEMORY;
+			goto exit;
+		}
+
+		tee_otp_get_hw_unique_key(&huk);
+		tee_otp_get_die_id(chip_id, sizeof(chip_id));
+
+		memcpy(message, chip_id, sizeof(chip_id));
+		memcpy(message + sizeof(chip_id), string_for_ssk_gen,
+				sizeof(string_for_ssk_gen));
+
+		res = generate_ssk(
```

Comment:
Will re-format the code like this:

		res = generate_ssk(enc_fs_ssk.key, sizeof(enc_fs_ssk.key),
				huk.data, sizeof(huk.data),
				message, sizeof(message));

Does it looks good for you, @jbech-linaro ?